### ISSUE TITLE
Few issues.

### ISSUE THREAD

**Issue by 60plus** [soda3x/ArmaReforgerServerTool#2]
Hi.
I don't know if you accept issues reporting
But I found a few of them.

1. After starting the application and server. App creates a file named mod_database.txt
    This file prevents the application from restarting when you close it. After deleting the file, the application restarts without a problem.

2. gameNumber value also missing under game: {} which makes impossible to run several server instances
![obraz](https://user-images.githubusercontent.com/73450566/172149092-a8fcb063-55ae-4838-b005-25febe870bca.png)

3. I can not connect to the servers that are created by application. I use exactly the same settings as the server that starts by the batch file and is working.
![obraz](https://user-images.githubusercontent.com/73450566/172153480-bf269d22-281f-4347-8949-784480012359.png)

When start by app server shows up on the list in the game browser but rejects connections.
![obraz](https://user-images.githubusercontent.com/73450566/172151561-a47077f8-f486-43d7-aac5-eb00062a2605.png)

![obraz](https://user-images.githubusercontent.com/73450566/172151208-ebd6f39f-e988-4640-9f17-c0203164050c.png)

**Comment by soda3x** [comment]
👋🏻 
These issues shouldn't be too hard to fix, thanks for bringing them to my attention.

1. The mod_database.txt issue I think can be fixed if I allow the user to configure where its saved. Are you perchance running the server tool from Program Files or some protected directory?
2. Game Number is pretty easy to implement, will get on that ASAP.
3. To me, "3" looks like another file permissions error, it's probably best run from your local app data folder as Windows applications are allowed to write there, try moving the application there and test "1" and "3" again if possible, would be great help!

Thanks for using my software, lets make it better!

**Comment by 60plus** [comment]
> The mod_database.txt issue I think can be fixed if I allow the user to configure where its saved. Are you perchance running the server tool from Program Files or some protected directory?

Im running the server tool from Desktop.

> To me, "3" looks like another file permissions error, it's probably best run from your local app data folder as Windows applications are allowed to write there, try moving the application there and test "1" and "3" again if possible, would be great help!

I moved server tool to local app data folder did not help same issues 1 nad 3. I copyed to another disks, folders but nothing helped.

**Comment by soda3x** [comment]
Ok I'll have a look into it. Thanks for trying!

**Comment by soda3x** [comment]
I've just released version 0.3.
Version 0.3 is now delivered using an installer, it will install to your Local App Data folder and you are able to specify where the server files are saved.
Hopefully some of your issues are resolved in this release. Let me know if not.

**Comment by 60plus** [comment]
Thanks for the update.
But unfortunately I have bad news. Now it's worse as it was. 

1. First issue solved. I can now restart the server tool after closing it.
2. gameNumber value dont change in server.json file created immediately after launch. No matter what I type in the gui is always 0
3. Server created by application always crash. (this may be due to issue 2) 
4. LOG 

> Steam Console Client (c) Valve Corporation - version 1654028333
-- type 'quit' to exit --
Loading Steam API...OK
Connecting anonymously to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
Success! App '1874900' already up to date.
Redirecting stderr to 'C:\Users\60plus\Documents\steamcmd\logs\stderr.txt'
[  0%] Wyszukiwanie dostępnych aktualizacji...
[----] Weryfikowanie instalacji...
ENGINE    (W): No module info used, trying to identify -> 0xD08A0000
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskVertical_256x1.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskVertical_256x1.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskHorizontal_256x1.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskHorizontal_256x1.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskCircular_1024.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskCircular_1024.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskCircular_1024.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ScreenEffects/Suppression/suppressionVignette_alphaMask200.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskCircular_1024.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskCircular_1024.edds'
GUI       (E): ImageWidget::AlphaMaskTexture can't load 'UI/Textures/ProgressMasks/ProgressMaskCircular_1024.edds'
BACKEND   (E): Unable to continue with a broken DS config!
RESOURCES (E): ==== Resource leaks ====
RESOURCES (E): ui/fonts/robotomono_msdf_28.edds   1
ENGINE       : Creating engine.
ENGINE       : FileSystem: Adding relative directory 'C:\Users\60plus\Documents\saves' to filesystem under name profile
ENGINE       : Initializing engine.
ENGINE       : CLI Params: -config C:\Users\60plus\Documents\server.json -profile C:\Users\60plus\Documents\saves -logStats 5000 
ENGINE       : FileSystem: Adding package './addons/data/' (pak count: 3) to filesystem under name ArmaReforger
ENGINE       : FileSystem: Adding package './addons/core/' (pak count: 1) to filesystem under name core
RESOURCES    : ResourceDB: loading cache (id=0 name=ArmaReforger path=./addons/data/resourceDatabase.rdb)
RESOURCES    : ResourceDB: loading cache (id=1 name=core path=./addons/core/resourceDatabase.rdb)
RESOURCES    : ResourceDB: loading cache (id=2 name=profile path=C:\Users\60plus\Documents\saves/resourceDatabase.rdb)
RENDER       : Basic screen setup:
RENDER       :   Back buffer width x height: 0x0
RENDER       :   Back buffer format	       : TEXFMT_UNKNOWN (0)
RENDER       :   MSAA                      : none
RENDER       : Detected 1 output devices:
RENDER       :   Dummy output (Dummy device for our dummy render.), 1920x1080 (offset 0x0)
RENDER       : GPU_DUMMY (0xFFFF) (Enfusion dummy render) with 1 GPU detected, VRAM 128MB (fast 128MB), shared 128MB, driver version 0, stereo not available
RENDER       :   Multithreaded resource creation is not supported
RENDER       :   Multithreaded command buffers are not supported
RENDER       :   Compute units are not supported
RENDER       :   Structured buffers are not supported
RENDER       :   Concurrent frames limit 2
RENDER       :   Shader precision support 16bit is not supported
ENGINE       : Job system settings: coreCount=6, shortJobContextCount=6, shortJobThreadCount=5, longJobThreadCount=2
AUDIO        : Audio device created - Channels: 2, Sample rate: 48000
ENGINE       : Initializing inputs.
NETWORK      : Initializing networking.
NETWORK      : Replication will be using RplNet.
SCRIPT       : Initializing scripts
 SCRIPT       : ScriptProjectManager init
 PROFILING    : ScriptProjectManager init took: 0.016400 ms
PROFILING    : Initializing scripts took: 47.161600 ms
ENGINE       : Enfusion engine successfully created.
GUI          : Using default language (en_us)
GUI          : Loading 'en_us' localization file.
INIT         : Loading StringTable
PROFILING    : Loading StringTable took: 17.772800 ms
SCRIPT       : Compiling GameLib scripts
PROFILING    : Compiling GameLib scripts took: 32.794300 ms
SCRIPT       : Compiling Game scripts
PROFILING    : Compiling Game scripts took: 3179.532300 ms
INIT         : Creating game instance(ArmaReforgerScripted), version 0.9.5.66 built 2022-06-02 17:46:53 UTC.
ENGINE       : Creating instance of Statistics manager
SCRIPT       : Profile - Using MachineName!
NETWORK      : Starting dedicated server using command line args.
BACKEND      : Required addons are ready to use.
DEFAULT      : Entered online game state.
ENGINE       : Game successfully created.
ENGINE       : Game destroyed.






5. I do not know what happened there but the old version had only one flag.
![obraz](https://user-images.githubusercontent.com/73450566/172424142-c505b0f5-660a-43b7-977f-9632de479499.png)

   and new one 
![obraz](https://user-images.githubusercontent.com/73450566/172424236-3b922ec0-ac19-4c1c-9b9c-5a362d383409.png)



Thanks for your work I appreciate very much. This tool would really be useful to the community.
If I can suggest something it is missing automatic restart at a certain time. Because the servers stop working after some time.

**Comment by soda3x** [comment]
Oh that sucks! It's working for me. I'll have another look later today.
I think I know what's wrong with the `gameNumber` value not working, easy fix.
As for the AV reports, I imagine that comes from it just being an NSIS installer now? They're definitely false-positives. The installer script is included in the repository for peace of mind.

**Comment by soda3x** [comment]
Hiya! I just released version 0.3.1, I had some people test it and it seems to be working correctly! Please give it a shot, hopefully you have some success with it 🤞. Let me know how it goes.

**Comment by soda3x** [comment]
Just install over the top of the previous version, no need to uninstall

**Comment by soda3x** [comment]
I've also just now implemented your automatic restart feature that you requested! Go check out version v0.4!

**Comment by soda3x** [comment]
That's great news, I'll check the ports later today

If you feel that way inclined, my PayPal is PayPal.me/bnewman227.
:)
Hope you're having a great day

**Comment by soda3x** [comment]
Hi mate, I've had a look at the ports issue and they seem to be working for me, ports are set correctly.
While the server is running, can you please check the generated `server.json` file located in the directory with the Arma Reforger Server files (this directory is referred to at the top of the Server Management window)

Please paste the contents of that file here.

**Comment by 60plus** [comment]
Hello sorry for the delay I had to do something and I did not have time to test and write back.

Values in the `server.json` file change. but the server rejects connections on ports other than 2001

![obraz](https://user-images.githubusercontent.com/73450566/173188970-47e16f5c-a248-4965-be9f-12c0eb8386b1.png)

![obraz](https://user-images.githubusercontent.com/73450566/173189092-f3280066-6ecc-4ce3-acaf-874c43f5486c.png)

![obraz](https://user-images.githubusercontent.com/73450566/173189192-f3d4488a-c33b-4f86-9580-bb653f294da3.png)

Ports are open and firewall is set.

![obraz](https://user-images.githubusercontent.com/73450566/173189306-2e8726a6-5aca-4e2b-94db-5fc52927ad11.png)

As soon as I change the port to 2001 server works as it should.I use the same ports when I run the server frome batch file and everything is ok.So it can not be the fault of my network or computer because exactly the same happens on 2 machines.

**Comment by soda3x** [comment]
What are the launch arguments you're using from the batch script?

**Comment by 60plus** [comment]
ArmaReforgerServer.exe -config ".\configs\configGMclean.json" -profile ArmaReforgerServer -maxFPS 60

**Comment by soda3x** [comment]
That's exactly the same arguments that is being used to start the server from the app.
I'll investigate some more and see if I can see something but basically, if it's in the json file, thats what is used in the server.

**Comment by 60plus** [comment]
Perhaps adding paths to own installation of the server and.json files help. instead of downloading a new from steam and create a new config.json ?

**Comment by soda3x** [comment]
That's more-or-less the same thing. Leave it with me, I'll sort it out :)

**Comment by soda3x** [comment]
Hi mate, so I did some investigating and found that the server app is definitely using the ports as expected. Attaching some pics to show you.

Configured server to use port "1337":
![Screenshot 2022-06-15 174414](https://user-images.githubusercontent.com/25409948/173777648-46a4f560-0412-43f0-9909-3f04bb4fae21.png)

Netstat gave me the following result. Notice the PID:
![Screenshot 2022-06-15 174438](https://user-images.githubusercontent.com/25409948/173777717-2acb6795-5cc7-4593-a78a-2835922570b4.png)

Details section of Task Manager shows that that PID is the Arma Server executable:
![Screenshot 2022-06-15 174449](https://user-images.githubusercontent.com/25409948/173777853-a6636726-2a72-4ef5-997a-f446b0cd37b2.png)

Not really sure what's going on on your end, but it should definitely be working.

**Comment by soda3x** [comment]
As for the laggy console, I'm working on that. No luck yet, but I'll keep you updated.

**Comment by 60plus** [comment]
> Hi mate, so I did some investigating and found that the server app is definitely using the ports as expected. Attaching some pics to show you.
> 
> Configured server to use port "1337": ![Screenshot 2022-06-15 174414](https://user-images.githubusercontent.com/25409948/173777648-46a4f560-0412-43f0-9909-3f04bb4fae21.png)
> 
> Netstat gave me the following result. Notice the PID: ![Screenshot 2022-06-15 174438](https://user-images.githubusercontent.com/25409948/173777717-2acb6795-5cc7-4593-a78a-2835922570b4.png)
> 
> Details section of Task Manager shows that that PID is the Arma Server executable: ![Screenshot 2022-06-15 174449](https://user-images.githubusercontent.com/25409948/173777853-a6636726-2a72-4ef5-997a-f446b0cd37b2.png)
> 
> Not really sure what's going on on your end, but it should definitely be working.



But can you connect to this server? For some reason I can connect only on port 2001 when I run the server from the application.
I started the server from folder where the application stores server files but with the batch file and my config and all ports work. As soon as I run through application only 2001 works. Everything is like on your screenshots but I can not connect.

![obraz](https://user-images.githubusercontent.com/73450566/173794832-2b3af14a-08ed-48a5-b14d-3bdd427a6944.png)

**Comment by soda3x** [comment]
Have you forwarded the ports on your Router?

**Comment by 60plus** [comment]
> Hello sorry for the delay I had to do something and I did not have time to test and write back.
> 
> Values in the `server.json` file change but not somewhere inside. Server rejects connections on ports other than 2001
> 
> ![obraz](https://user-images.githubusercontent.com/73450566/173188970-47e16f5c-a248-4965-be9f-12c0eb8386b1.png)
> 
> ![obraz](https://user-images.githubusercontent.com/73450566/173189092-f3280066-6ecc-4ce3-acaf-874c43f5486c.png)
> 
> ![obraz](https://user-images.githubusercontent.com/73450566/173189192-f3d4488a-c33b-4f86-9580-bb653f294da3.png)
> 
> Ports are open and firewall is set.
> 
> ![obraz](https://user-images.githubusercontent.com/73450566/173189306-2e8726a6-5aca-4e2b-94db-5fc52927ad11.png)
> 
> ![obraz](https://user-images.githubusercontent.com/73450566/173189775-9b37b215-9227-4e3c-a757-3a0f69b77b01.png)
> 
> As soon as I change the port to 2001 server works as it should.So it can not be the fault of my network or computer because exactly the same happens on 2 machines.
> 
> I use the same ports when I run the server frome batch file and everything is ok ![obraz](https://user-images.githubusercontent.com/73450566/173190235-2ae4cb3a-0150-4818-b36b-de50149f0551.png) ![obraz](https://user-images.githubusercontent.com/73450566/173193151-f65c790a-73a1-48c9-b498-39d43ddbc442.png)

Yeah i did and firewall is open. Server works with bach file and i can connect no problem.

**Comment by soda3x** [comment]
Nah, I'll get it working!

**Comment by 60plus** [comment]
Similar problem occurs in most Reforger servers in docker. I found only one where you can change the port and server works but lacks many env values and the owner is unwilling to cooperate. exactly speaking took the repo out from github :(

**Comment by soda3x** [comment]
OK so for the time being, I will add a "Force Port" option, this will be in v0.4.2. Stay tuned.

**Comment by soda3x** [comment]
Just uploaded v0.4.2. Let me know how it goes!

**Comment by 60plus** [comment]
I apologize for the delay.Bohemia decided to do update Reforger Server and Game during testing.

Combination of settings :

> Server Configuration

`Game Host Bind Port` and `Register Bind Port` set to 2001

> Server Management

`Override Port` set to selected port (2002 or 20001 in my case)

Makes the server look like is working and I can connect. But soon when i switch port 2001 off in my router I can not connect again.
Which means that the server is still running on port 2001.Regardless of what is written in the console , Override Port or server.json. 
What confirmed me in this belief is the fact that after starting second instance of the server with changed ports , name and steam query. Server appears in the browser but tries to connect with the first instance and not with the second.

**Comment by soda3x** [comment]
I just found a bug that might be the cause of the Override Port option not working! Making a hotfix now

**Comment by soda3x** [comment]
v0.4.3 is up!

**Comment by 60plus** [comment]
Do not breathe or move. Don't touch anything. It works !!! 😂
Whatever you did it worked. You are a genius !!!!

Server run on port 20001

![obraz](https://user-images.githubusercontent.com/73450566/174275102-65701a9c-6501-4f0b-aa27-25cba61fe684.png)

and port 2001 is off

![obraz](https://user-images.githubusercontent.com/73450566/174275401-c4daab18-a893-4d45-b6eb-b3d2e7c5f3a5.png)


I created a post on steam with a link to this tool. I hope you don't mind and that people will find it as helpful as I did.
Thank you very much for your patience and time.
